contracts/MiniMeToken.sol

```diff
@@ -281,6 +312,99 @@ contract MiniMeToken is Controlled {
         return totalSupplyAt(block.number);
     }
 
+////////////////
+// Permit and transferWithAuthorization
+////////////////
+    function permit(
+        address _owner,
+        address _spender,
+        uint256 _value,
+        uint256 _deadline,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) public {
+        require(_deadline >= block.timestamp, "permit: AUTH_EXPIRED");
+
+        bytes32 encodeData = keccak256(
+            abi.encode(
+                PERMIT_TYPEHASH,
+                _owner,
+                _spender,
+                _value,
+                nonces[_owner]++,
+                _deadline
+            )
+        );
+        _validateSignedData(_owner, encodeData, _v, _r, _s);
+        _approve(_owner, _spender, _value);
+    }
+
+    function getDomainSeparator() public constant returns (bytes32) {
```

Comment:
Was wondering if it would be better to have a `bytes32 public DOMAIN_SEPARATOR` instead and set it in the constructor?

Reply:
You'd like to think because it's labelled `constant` as opposed to `view` it would effectively do what you're suggesting at compile time. However, I don't think the `constant` keyword in this version of Solidity is any different to using `view`, I believe it's just for readability. Once `pure` was introduced I think this changed. So I think it probably is worth doing, since we're deploying to more resource constrained environments, unless this particular function is required by any of the standards, which I can't tell from the mentioned links. @eduadiez What are your thoughts?

Reply:
This could be said to be inherited from the implementation of the Aragon Token v2 and the HEZ token. The reasoning behind this is that the `chainID` needs to be calculated at runtime, otherwise replay-attacks could occur when faced with a network hardfork. Since this version of solidity does not support obtaining the `chainId` dynamically we had to hardcode it so this no longer makes the same sense.

https://github.com/aragon/aragon-network-token/blob/3b6bfcb7068f99e0d67ac9cf2e3115ac0d13c63a/packages/v2/contracts/ANTv2.sol#L104

Since `view` was introduce later that's why we need to use `constant`.

I don't see any inconvenience in changing it if you think it may provide some benefit, but for consistency I would leave it that way

Reply:
`view` actually is available in this version of Solidity. Whether it treats the function as constant or not I'm unsure but after running a gas test it doesn't seem to make any difference using one over the other.

Let's just leave as is, the saving would be negligible.